lib/miq_automation_engine/engine/miq_ae_method_service/miq_ae_service_model_base.rb

```diff
@@ -361,5 +361,9 @@ def self.ar_method
     def ar_method(&block)
       self.class.ar_method(&block)
     end
+
+    def ==(other)
+      self.id == other.id && self.class == other.class
```

Comment:
@fdupont-redhat I worked out the issue with the failing tests.  Not all objects expose an `id` method so this will raise an error in the first part of the comparison.  You can avoid this by checking the class first.  And you can drop the `self.` from the `id`.

`self.class == other.class && id == other.id`